Fix issues where Dropbox launches but the directory does not open

mariospr opened the pull request:
https://phabricator.endlessm.com/T22823

ramcq commented on src/eos-dropbox-app.py:
```diff
@@ -252,35 +259,34 @@ def _setup_config_monitor(self):
+        # The launcher and daemon processes might take a while to be spawned
+        # on slower machines, so we grant ourselves some room to wait for them.
+        for i in range(MAX_ATTEMPTS):
+            self._daemon = find_dropbox_daemon()
```
Relying on the fact the daemon process now replaces the launcher process with exec, I think this could be simplified and we can just ignore the launcher, and directly start the script and then track it as the daemon with popen.wait(). Should make things a bit simpler / less fragile.

mariospr replied:
I think you're right, I think I got a bit too stubborn trying to reuse the thread I setup before, it's probably no longer needed. Will update it then.

Commit:
eos-dropbox-app: Fix monitoring of the dropbox daemon

This was previously implemented by waiting for the launcher to die
and then finding the daemon's process to monitor it, but this does
not work anymore since now the original launcher does not really die,
but gets replaced with the daemon's process... meaning that the
monitoring is never setup until the original process exits.

So, simplify everything a bit by simply relying in the blocking call
to Popen and then .wait() afterwards to know when to kill the flatpak
process after the daemon dies, and get rid of so many weird things we
added in the past to deal with the previous situation.

https://phabricator.endlessm.com/T22823
